e2e/tests/eth_integration.rs

```diff
@@ -0,0 +1,273 @@
+use web3::signing::SecretKeyRef;
+
+mod ganache;
+#[macro_use]
+mod services;
+use crate::services::{
+    create_orderbook_api, deploy_mintable_token, to_wei, GPv2, OrderbookServices, UniswapContracts,
+    API_HOST,
+};
+
+const TRADER_BUY_ETH_A_PK: [u8; 32] =
+    hex!("0000000000000000000000000000000000000000000000000000000000000001");
+const TRADER_BUY_ETH_B_PK: [u8; 32] =
+    hex!("0000000000000000000000000000000000000000000000000000000000000002");
+const TRADER_SELL_ETH_PK: [u8; 32] =
+    hex!("0000000000000000000000000000000000000000000000000000000000000003");
+
+const ORDER_PLACEMENT_ENDPOINT: &str = "/api/v1/orders/";
+const FEE_ENDPOINT: &str = "/api/v1/fee/";
+
+#[tokio::test]
+async fn ganache_eth_integration() {
+    ganache::test(eth_integration).await;
+}
+
+async fn eth_integration(web3: Web3) {
+    shared::tracing::initialize("warn,orderbook=debug,solver=debug");
+    let chain_id = web3
+        .eth()
+        .chain_id()
+        .await
+        .expect("Could not get chainId")
+        .as_u64();
+
+    let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+    let solver = Account::Local(accounts[0], None);
+    let trader_buy_eth_a =
+        Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_A_PK).unwrap(), None);
+    let trader_buy_eth_b =
+        Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_B_PK).unwrap(), None);
+    let trader_sell_weth =
+        Account::Offline(PrivateKey::from_raw(TRADER_SELL_ETH_PK).unwrap(), None);
+
+    let gpv2 = GPv2::fetch(&web3, &solver).await;
+    let UniswapContracts {
+        uniswap_factory,
+        uniswap_router,
+    } = UniswapContracts::fetch(&web3).await;
+
+    // Create & Mint tokens to trade
+    let token = deploy_mintable_token(&web3).await;
+    tx!(solver, token.mint(solver.address(), to_wei(100_000)));
+    tx!(solver, token.mint(trader_buy_eth_a.address(), to_wei(50)));
+    tx!(solver, token.mint(trader_buy_eth_b.address(), to_wei(50)));
+
+    let weth = WETH9::builder(&web3)
+        .deploy()
+        .await
+        .expect("WETH deployment failed");
+    tx_value!(solver, to_wei(100_000), weth.deposit());
+    tx_value!(solver, to_wei(100), weth.deposit());
+    tx!(
+        solver,
+        weth.transfer(trader_sell_weth.address(), to_wei(100))
+    );
+
+    // Create and fund Uniswap pool
+    tx!(
+        solver,
+        uniswap_factory.create_pair(token.address(), weth.address())
+    );
+    tx!(
+        solver,
+        token.approve(uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        weth.approve(uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        uniswap_router.add_liquidity(
+            token.address(),
+            weth.address(),
+            to_wei(100_000),
+            to_wei(100_000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
```

Comment:
Maybe we can share these lines in a util method across tests (could also include the required minting).